src/python/pants/backend/jvm/tasks/jar_task.py

```diff
@@ -348,12 +348,13 @@ def __init__(self, context, jar):
       self._jar = jar
       self._manifest = Manifest()
 
-    def add_target(self, target, recursive=False):
+    def add_target(self, target, recursive=False, exclude_jar_library=False):
```

Comment:
As noted on the issue - I think this should be treated as a plain and simple regression and the underlying ClasspathUtil + its use corrected.